autocpenv.py

```diff
@@ -82,6 +82,11 @@ def OnJobSubmitted(self, job):
             (self.resolve_from_job_plugin, (plugin_mapping, job_plugin)),
         )
 
+        forced_requirements = self.resolve_from_job_plugin(self.GetConfigEntry('forced_plugin_mapping'), job_plugin)
+        self.log("Forced requirements (these will be appended to the job requirements):")
+        self.log(" ".join(forced_requirements))
+        requirements.extend(forced_requirements)
```

Comment:
This call to extend is likely to raise a TypeError because the `resolve_from_job_plugin` will return None if the `job_plugin` is not found in the mapping. So, make sure you check `forced_requirements` before extending the list of requirements.

Also, I think you should check each forced requirement to make sure it's not already in the requirements list before adding it. Cpenv doesn't prevent you from activating multiple versions of the same module, so checking here is a good way to guarantee jobs get a valid Environment.